apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -1,65 +1,109 @@
+import { Cached } from '../../../shared/interceptors';
+import { CacheKeyPrefixEnum } from '../../../shared/services/cache';
+
+const LOG_CONTEXT = 'TriggerEventUseCase';
 
 @Injectable()
 export class TriggerEvent {
   constructor(
-    private notificationTemplateRepository: NotificationTemplateRepository,
+    private addJobUsecase: AddJob,
+    private createNotification: CreateNotification,
     private processSubscriber: ProcessSubscriber,
     private jobRepository: JobRepository,
-    private addJobUsecase: AddJob,
     private notificationRepository: NotificationRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
     protected createExecutionDetails: CreateExecutionDetails
   ) {}
 
   async execute(command: TriggerEventCommand) {
-    await this.validateTransactionIdProperty(command.transactionId, command.organizationId, command.environmentId);
+    const { actor, environmentId, identifier, organizationId, to, userId } = command;
 
-    const template = await this.notificationTemplateRepository.findByTriggerIdentifier(
-      command.environmentId,
-      command.identifier
-    );
+    await this.validateTransactionIdProperty(command.transactionId, organizationId, environmentId);
 
     Sentry.addBreadcrumb({
       message: 'Sending trigger',
       data: {
-        triggerIdentifier: command.identifier,
+        triggerIdentifier: identifier,
       },
     });
 
+    const template = await this.notificationTemplateRepository.findByTriggerIdentifier(
+      command.environmentId,
+      command.identifier
+    );
+
+    /*
+     * Makes no sense to execute anything if template doesn't exist
+     * TODO: Send a 404?
+     */
+    if (!template) {
+      const message = 'Notification template could not be found';
+      Logger.error(message, LOG_CONTEXT);
+      throw new ApiException(message);
+    }
+
     const jobs: Omit<JobEntity, '_id'>[][] = [];
 
-    for (const subscriberToTrigger of command.to) {
-      jobs.push(
-        await this.processSubscriber.execute(
-          ProcessSubscriberCommand.create({
-            identifier: command.identifier,
-            payload: command.payload,
-            overrides: command.overrides,
-            to: subscriberToTrigger,
-            transactionId: command.transactionId,
-            environmentId: command.environmentId,
-            organizationId: command.organizationId,
-            userId: command.organizationId,
-            templateId: template._id,
-            actor: command.actor,
-            template,
-          })
-        )
+    // We might have a single actor for every trigger so we only need to check for it once
+    let actorProcessed;
+    if (actor) {
+      actorProcessed = await this.processSubscriber.execute(
+        ProcessSubscriberCommand.create({
+          environmentId,
+          organizationId,
+          userId,
+          subscriber: actor,
+        })
+      );
+    }
```

Comment:
Before, when sending the actor, we were searching for it in the DB inside the loop. Therefore we would be searching for the actor _number of subscribers times_. Not efficient. 😅